jsonargparse/_typehints.py

```diff
@@ -904,6 +939,16 @@ def adapt_typehints(
         elif typehint_origin is OrderedDict:
             val = dict(val) if serialize else OrderedDict(val)
 
+    # NotRequired
+    elif typehint_origin in not_required_types:
+        assert len(subtypehints) == 1, "NotRequired requires a single type argument"
```

Comment:
Why are this assert and the one below needed? I tried to create in python a `NotRequired` type hint without an argument and with more than one argument, and it is simply not allowed. So it seems this will never happen and there is no need for the assert.

Reply:
I meant it to serve more as a note to a maintainer as a condition that should never happen, thus justifying the use of the `0` indexing operation in the following line without checking that `subtypehints` is empty or not.

Reply:
Fair enough. Then lets leave it like this.